### PR TITLE
Add Poisson and geometric distributions and associated tests

### PR DESCRIPTION
**Pull request by tramy1258**
## Checklist

- [x] My pull request has a clear and explanatory title.
- [x] If necessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [x] I added appropriate unit tests.
- [ ] I made sure the code passes all unit tests. (refer to comment below)
- [x] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [x] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#coding-style-guidelines) and API.
- [x] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#documentation))
- [ ] I linked to issues and PRs that are relevant to this PR.

### CODE REVIEW

File: geomstats/information_geometry/poisson.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+            ----------
+            k : array-like, shape=[n_points,]
+                Point representing an Poisson distribution (lambda).
+
+            Returns
+            -------
+            pmf_at_k : array-like, shape=[..., n_points]
+                Probability mass function of the Poisson distribution with
+                parameters provided by point.
+            """
```
Comment (alebrigant) [inline geomstats/information_geometry/poisson.py:152]: Same comments as for the geometric distributions: samples and not points representing Poisson distributions.

File: geomstats/information_geometry/poisson.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+            If None, an end point must be given and a logarithm is computed.
+
+        Returns
+        -------
+        path : callable
+            Time parameterized geodesic curve. If a batch of initial
+            conditions is passed, the output array's first dimension
+            represents time, and the second corresponds to the different
+            initial conditions.
+        """
```
Comment (alebrigant) [inline geomstats/information_geometry/poisson.py:316]: Same comments as for geometric distributions.

File: geomstats/information_geometry/geometric.py [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+    def __init__(self):
+        super().__init__(dim=1)
+
+    def squared_dist(self, point_a, point_b, **kwargs):
+        """Compute squared distance associated with the geometric metric.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., 1]
+            Point representing an geometric distribution (lambda parameter).
```
Comment (alebrigant) [inline geomstats/information_geometry/geometric.py:181]: lambda is non standard for the parameter of a geometric distribution, I'm not sure it helps.

File: tests/data/geometric_data.py [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+                -(gs.arctanh(gs.sqrt(1 - 1e-6)) - gs.arctanh(gs.sqrt(1 - base_point)))
+                * 2
+                * base_point
+                * gs.sqrt(1 - base_point)
+            )
+            random_vec = gs.where(
+                (random_vec > min_tangent_vec) & (random_vec < max_tangent_vec),
+                random_vec,
+                random_vec % (max_tangent_vec - min_tangent_vec) + min_tangent_vec,
+            )
```
Comment (alebrigant) [inline tests/data/geometric_data.py:197]: What is this for ? Maybe add a word on this is the docstring of the test_data ?

File: geomstats/information_geometry/geometric.py [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+            Optional, default: None.
+            If None, an end point must be given and a logarithm is computed.
+
+        Returns
+        -------
+        path : callable
+            Time parameterized geodesic curve. If a batch of initial
+            conditions is passed, the output array's first dimension
+            represents time, and the second corresponds to the different
+            initial conditions.
```
[the remark this answers is not in the document]
Comment (alebrigant) [reply geomstats/information_geometry/geometric.py:328]: Is the output shape of `geodesic` tested ? It should be enough to test the output shape of `_geodesic_ivp` and `_geodesic_bvp` through the shapes of `exp` and `log`, but I don't see these tests (test_exp_shape and test_log_shape) in the test file.

**Review by alebrigant — CHANGES_REQUESTED** [review]
Nice ! Most comments concern the docstrings. I'm also not sure if the output shape of geodesics is right. The first dimension should always correspond to vectorization, so usually it is `[n_points, n_times, dim]`. Since here we are in dimension 1, is the output shape `[n_points, n_times, 1]` or `[n_points, n_times]` ?

File: geomstats/information_geometry/geometric.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tramy1258) [reply geomstats/information_geometry/geometric.py:328]: Yes, the shapes are tested. In `test_geometric`, `TestGeometricMetric` inherits `RiemannianMetricTestCase` which inherits `ConnectionTestCase` which contains tests for `exp` and `log`. It is actually just the docstring that is incorrect.

File: geomstats/information_geometry/geometric.py [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+        def pmf(k):
+            """Generate parameterized function for geometric pmf.
+
+            Compute the probability mass function of the geometric
+            distribution with parameters provided by point.
+
+            Parameters
+            ----------
+            k : array-like, shape=[n_points,]
+                Point representing an geometric distribution (lambda).
```
[the remark this answers is not in the document]
Comment (alebrigant) [reply geomstats/information_geometry/geometric.py:149]: Replace also "Point representing a geometric distribution" by "Point in the sample space, i.e. an integer."

File: geomstats/information_geometry/poisson.py [hunk 7]
```diff
@@ -141,12 +141,12 @@ def pmf(k):
 
             Parameters
             ----------
-            k : array-like, shape=[n_points,]
+            k : array-like, shape=[n_samples,]
                 Point representing an Poisson distribution (lambda).
```
Comment (alebrigant) [inline geomstats/information_geometry/poisson.py:145]: Change to "Point in the sample space, i.e. an integer."

**Review by alebrigant — APPROVED** [review]
Thanks for the changes ! Last very minor docstring changes and we are good to merge !

**Comment by alebrigant** [comment]
@tramy1258 sorry last request: I didn't see it before but there are some remaining flake 8 errors (see Linting). Could you please fix these before the merge ? Almost there !

**Comment by tramy1258** [comment]
> @tramy1258 sorry last request: I didn't see it before but there are some remaining flake 8 errors (see Linting). Could you please fix these before the merge ? Almost there !

It is all done normally!

**Pull request merged**